plugins/node/opentelemetry-hapi-instrumentation/src/hapi.ts

```diff
@@ -357,26 +361,30 @@ export class HapiInstrumentation extends BasePlugin<typeof Hapi> {
     const instrumentation: HapiInstrumentation = this;
     if (route[handlerPatched] === true) return route;
     route[handlerPatched] = true;
-    if (typeof route.handler === 'function') {
-      const handler = route.handler as Hapi.Lifecycle.Method;
+    const oldHandler = route.options?.handler ?? route.handler;
+    if (typeof oldHandler === 'function') {
       const newHandler: Hapi.Lifecycle.Method = async function (
         request: Hapi.Request,
         h: Hapi.ResponseToolkit,
         err?: Error
       ) {
         if (instrumentation._tracer.getCurrentSpan() === undefined) {
-          return await handler(request, h, err);
+          return await oldHandler(request, h, err);
         }
         const metadata = getRouteMetadata(route, pluginName);
         const span = instrumentation._tracer.startSpan(metadata.name, {
           attributes: metadata.attributes,
         });
-        const res = await handler(request, h, err);
+        const res = await oldHandler(request, h, err);
         span.end();
 
         return res;
       };
-      route.handler = newHandler;
+      if (route.handler) {
```

Comment:
Nit: to keep the same order as in line 364 I think it should be
```
if (route.options?.handler) {
  route.options.handler = newHandler
} else {
  route.handler = newHandler;
}
```

Reply:
Yea sure, I will make the change